redis_commands.c

```diff
@@ -3911,12 +3911,15 @@ void redis_setoption_handler(INTERNAL_FUNCTION_PARAMETERS,
                 zend_string_release(redis_sock->prefix);
                 redis_sock->prefix = NULL;
             }
-            if (val_str && val_len > 0) {
-                redis_sock->prefix = zend_string_init(val_str, val_len, 0);
+            val_str = zval_get_string(val);
+            if (ZSTR_LEN(val_str) > 0) {
```

Comment:
@jrchamp Do we need this check here?

Reply:
Sorry - I'm not as familiar with the zend code.